src/components/src/ImageContainer/image.container.tsx

```diff
@@ -19,69 +19,81 @@
   imgUrl: string;
-  /** class name */
-  className?: string;
-  /** is a sku image */
+  /** picture element class name */
+  pictureClassName?: string;
+  /** img element class name */
+  imgClassName?: string;
+  /** is a sku image, determines whether to use `skuImagesUrl` or `siteImagesUrl` in ep.config.json as url prefix in srcSet */
   isSkuImage?: boolean;
-  /** loading data */
-  onLoadData?: (...args: any[]) => any;
+  /** Types to use for srcSet */
+  types?: string[];
+  /** width breakpoints to serve different images */
+  sizes?: string[];
+  /** alt to be passed to img element */
+  alt: string,
+  /** The onLoadData */
+  onLoadData?: any,
 }
 
 function ImageContainer(props: ImageContainerProps) {
   const {
-    fileName, imgUrl, className, isSkuImage, onLoadData,
+    imgUrl, pictureClassName, imgClassName, isSkuImage, types, sizes, fileName, alt, onLoadData,
   } = props;
 
-  const imageSource = isSkuImage ? Config.skuImagesUrl.replace('%sku%', fileName) : Config.siteImagesUrl.replace('%fileName%', fileName);
+  const imageSizes = sizes === undefined ? Config.ImageContainerSrcs.sizes : sizes;
+  const imageTypes = types === undefined ? Config.ImageContainerSrcs.types : types;
+  let imgPrefix = '';
+  const imgAlt = alt != null ? alt : '';
+
+  if (isSkuImage) {
+    imgPrefix = Config.skuImagesUrl;
+  } else {
+    imgPrefix = Config.siteImagesUrl;
+  }
+
+  const [error, setError] = useState(false);
 
   const handleError = (e, defaultImgUrl) => {
-    const { src } = e.target;
-    if (Config.imageFileTypes && Config.imageFileTypes.enable && e.target['data-source'] !== 'default') {
-      const fallbackTypes = Config.imageFileTypes.types;
-      const initType = imgUrl.replace(/.*(?=\.)/g, '');
-      const types = (fallbackTypes && fallbackTypes.length > 0) ? [
-        initType,
-        ...fallbackTypes.filter(fileType => fileType !== initType),
-      ] : [];
-      const currentType = src.replace(/.*(?=\.)/g, '');
-      const i = types.indexOf(currentType);
-      if (types[i + 1]) {
-        e.target.src = src.replace(currentType, types[i + 1]);
-      } else {
-        e.target.src = isSkuImage ? imgPlaceholder : defaultImgUrl;
-        e.target['data-source'] = 'default';
-      }
-    } else if (e.target['data-source'] !== 'default') {
-      e.target.src = isSkuImage ? imgPlaceholder : defaultImgUrl;
-      e.target['data-source'] = 'default';
-    }
+    e.currentTarget.src = isSkuImage ? imgPlaceholder : defaultImgUrl;
+    setError(true);
   };
 
-  return (
-    <img
-      className={className}
-      alt=""
-      src={imageSource}
-      onError={e => handleError(e, imgUrl)}
-      onLoad={onLoadData}
-    />
-  );
+  const generateSrcSet = type => imageSizes.reduce((acc, imageSize) => {
+    if (acc === '') {
+      return `${acc}${imgPrefix.replace('%fileName%', `${type}/${fileName}-${imageSize}w.${type} ${imageSize}w`)}`;
```

Comment:
`acc` is an empty string so you don't need it here. Whole if can be turned into a ternary operator.

Reply:
Actually, this is better to be implemented using `imageSizes.map(...).join(', ')` instead of `.reduce`